### PR TITLE
sc2: Adding larva command card submenu for advanced strains

### PR DESCRIPTION
**Pull request by MatthewMarinets**
also moving pygalisks to larva

Made description and logic changes to pygalisk in [client PR #364](https://github.com/Ziktofel/Archipelago/pull/364)

Tentative new layout:
image
(scourge hotkey changed locally S -> X, the rest should all be default)

The submenu is sticky, so multiple advanced morphs can be built after opening the submenu once.
larva_submenus

I hope this opens the door to more basic morphs, and possibly some "morph the advanced unit straight from larva" upgrades, particularly for banelings (AI gets them sometimes, would get me to build them more) and Vipers (melee-style).

### TODO
- [ ] Update Pygalisk unlock trigger to give spawning pool and not give predator nest / merc upgrades
- [ ] Investigate larva cancel smart command card; if it never works, remove it and move submenu controls to 0,4 instead of 0,3

### CODE REVIEW

**Comment by MatthewMarinets** [comment]
Added the gamestrings for the submenu buttons. Note the submenu buttons are available immediately, even if no advanced morphs are unlocked.
image
image

**Comment by MatthewMarinets** [comment]
Added a flag so the advanced morphs subtab only shows up if you have a unit in there.
image
image

Unfortunately, it greys the submenu out if you don't have the requirements for any unit in the subtab :/
image

**Comment by MatthewMarinets** [comment]
Fixed the greyed-out submenu button issue.
image
image
image

It involved manually adding all advanced morphs to a requirement, which is unfortunate as it's an easy thing to forget if adding new advanced morphs in future. I think it's easy enough to spot and fix if we do add more, though

File: Mods/ArchipelagoPlayer.SC2Mod/enUS.SC2Data/LocalizedData/GameHotkeys.txt [hunk 1]
```diff
@@ -196,6 +196,8 @@ Button/Hotkey/AP_K5Leviathan=V
 Button/Hotkey/AP_KerriganUltimateAbilites=U
 Button/Hotkey/AP_KhaydarinMonolith=D
 Button/Hotkey/AP_Lair=L
+Button/Hotkey/AP_LarvaAdvancedMorphs=X
+Button/Hotkey/AP_LarvaBasicMorphs=B
```
Comment (Ziktofel) [inline Mods/ArchipelagoPlayer.SC2Mod/enUS.SC2Data/LocalizedData/GameHotkeys.txt:200]: Escape should be the default hotkey for exiting a submenu

**Pull request merged**